### PR TITLE
Add linux osx

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by jdblischak** [comment]
@xylar What is the status of this PR? The build logs are no longer available, so I can't easily view the error messages. I'm interested in building the [Azure SDK for C++](https://github.com/Azure/azure-sdk-for-cpp) libraries for conda-forge, and this is a recursive dependency: `azure-core` -> `azure-uamqp-c` -> `azure-c-shared-utility`

**Comment by jdblischak** [comment]
I ran this branch on my fork.

The [linux-64 build failed](https://dev.azure.com/jdblischak/feedstock-builds/_build/results?buildId=349&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=185) because it couldn't find a C++ compiler (at least that is my interpretation of the error message):

```
-- The C compiler identification is GNU 10.4.0
-- The CXX compiler identification is unknown
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at CMakeLists.txt:10 (project):
  No CMAKE_CXX_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
```

The build completed for the [osx-64 job](https://dev.azure.com/jdblischak/feedstock-builds/_build/results?buildId=349&view=logs&j=58ac6aab-c4bc-5de2-2894-98e408cc8ec9&t=933f325c-924e-533d-4d95-e93b5843ce8b&l=723) but then failed the test:

```
+ test -f /Users/runner/miniforge3/conda-bld/azure-c-shared-utility_1687376730560/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/aziotsharedutil.dylib
Tests failed for azure-c-shared-utility-2020.01.22-h26f4a9b_1.conda - moving package to /Users/runner/miniforge3/conda-bld/broken

```

**Comment by xylar** [comment]
@conda-forge-admin, please restart ci

**Comment by xylar** [comment]
@jdblischak, I believe the status of this PR is mostly that I realized just what a can of worms azure's whole non-python package stask is and that is was both over my head and would take more time than I have.  If you want to revive this work, the simplest way forward might be to open a new PR using whatever you want to keep from this one.  That way, you would have full control.

**Comment by jdblischak** [comment]
>  I realized just what a can of worms azure's whole non-python package stask is and that is was both over my head and would take more time than I have.

Oh jeez. That's not encouraging. You are clearly more knowledgeable than I am in regards to C++. I currently have the motivation/need though, so we'll see where I can get

> If you want to revive this work, the simplest way forward might be to open a new PR using whatever you want to keep from this one. That way, you would have full control.

That works for me. If it's ok, I'll probably ping you occasionally for advice

**Comment by jdblischak** [comment]
Quick update: I was able to fix the build on macOS by updating the path to the expected file in the test:

```
    - test -f "${PREFIX}/lib/libaziotsharedutil${SHLIB_EXT}"  # [not win]
```

And including `{{ compiler('cxx') }}` as a build requirement for the linux build fixed the CMake error. However, now it is failing during the compilation step. It appears to be related to openssl, but my attempt to pin openssl to 1.1.1 wasn't able to fix it. It may require updating to [LTS_07_2022_Ref02](https://github.com/Azure/azure-c-shared-utility/releases/tag/LTS_07_2022_Ref02), which added openssl 3 compatibility

**Comment by jjerphan** [comment]
Like @xylar, I also think the Azure SDK for C++ libraries' suite is hard to distribute.

Let's rerun the CI to have the logs and understand problems you met.

**Comment by jjerphan** [comment]
@conda-forge-admin, please restart ci

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,26 +10,28 @@ source:
...
-  skip: true  # [not win64]
+  number: 1
   run_exports:
-    - {{ pin_subpackage(name, exact=True) }}
+    - {{ pin_subpackage(name, max_pin='x.x.x') }}
 
 requirements:
   build:
-    - vs2017_win-64          # [win64]
+    - {{ compiler('c') }}
```
Comment (jjerphan) [inline recipe/meta.yaml:19]: Both compilers are needed.
```suggestion
    - {{ compiler('c') }}
    - {{ compiler('cxx') }}
```

**Comment by jdblischak** [comment]
@jjerphan thanks for your interest in helping out with this PR

> Let's rerun the CI to have the logs and understand problems you met.

However, I don't want you to waste time debugging the same errors that I already fixed (https://github.com/conda-forge/azure-c-shared-utility-feedstock/pull/4#issuecomment-1603065392). As @xylar requested above, I started working on this on my fork

https://github.com/jdblischak/azure-c-shared-utility-feedstock/commits/add_linux_osx_jdb

I fixed the [missing compiler](https://github.com/jdblischak/azure-c-shared-utility-feedstock/commit/d3e788e5ab843de4d854025b3d53a5bdf3c5c439) and the [incorrect filename in the test section](https://github.com/jdblischak/azure-c-shared-utility-feedstock/commit/546af2c0bf05826f51228163891df3b5a4abb746).

Unfortunately my build logs have already been deleted, so I just restarted a new one to run on my "add_linux_osx_jdb" branch

https://dev.azure.com/jdblischak/feedstock-builds/_build/results?buildId=431&view=results

**Comment by jdblischak** [comment]
@jjerphan here's the error message for the linux-64 build:

```sh
[ 80%] Building C object CMakeFiles/aziotsharedutil.dir/adapters/tlsio_openssl.c.o
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c: In function 'add_certificate_to_store':
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:872:24: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  872 |             bio_method = BIO_s_mem();
      |                        ^
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c: In function 'create_openssl_instance':
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:943:9: error: 'TLSv1_2_method' is deprecated: Since OpenSSL 1.1.0 [-Werror=deprecated-declarations]
  943 |         method = TLSv1_2_method();
      |         ^~~~~~
In file included from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:5:
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/ssl.h:2034:50: note: declared here
 2034 | OSSL_DEPRECATEDIN_1_1_0 __owur const SSL_METHOD *TLSv1_2_method(void); /* TLSv1.2 */
      |                                                  ^~~~~~~~~~~~~~
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:947:9: error: 'TLSv1_1_method' is deprecated: Since OpenSSL 1.1.0 [-Werror=deprecated-declarations]
  947 |         method = TLSv1_1_method();
      |         ^~~~~~
In file included from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:5:
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/ssl.h:2026:50: note: declared here
 2026 | OSSL_DEPRECATEDIN_1_1_0 __owur const SSL_METHOD *TLSv1_1_method(void); /* TLSv1.1 */
      |                                                  ^~~~~~~~~~~~~~
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:951:9: error: 'TLSv1_method' is deprecated: Since OpenSSL 1.1.0 [-Werror=deprecated-declarations]
  951 |         method = TLSv1_method();
      |         ^~~~~~
In file included from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:5:
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/ssl.h:2018:50: note: declared here
 2018 | OSSL_DEPRECATEDIN_1_1_0 __owur const SSL_METHOD *TLSv1_method(void); /* TLSv1.0 */
      |                                                  ^~~~~~~~~~~~
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c: In function 'tlsio_openssl_init':
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:1066:5: error: 'ERR_load_BIO_strings' is deprecated: Since OpenSSL 3.0 [-Werror=deprecated-declarations]
 1066 |     ERR_load_BIO_strings();
      |     ^~~~~~~~~~~~~~~~~~~~
In file included from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/cryptoerr.h:17,
                 from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/crypto.h:38,
                 from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/comp.h:22,
                 from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/ssl.h:28,
                 from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:5:
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/cryptoerr_legacy.h:31:27: note: declared here
   31 | OSSL_DEPRECATEDIN_3_0 int ERR_load_BIO_strings(void);
      |                           ^~~~~~~~~~~~~~~~~~~~
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c: In function 'tlsio_openssl_deinit':
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:1094:5: error: 'ERR_remove_thread_state' is deprecated: Since OpenSSL 1.1.0 [-Werror=deprecated-declarations]
 1094 |     ERR_remove_thread_state(NULL);
      |     ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/work/adapters/tlsio_openssl.c:6:
/home/conda/feedstock_root/build_artifacts/azure-c-shared-utility_1691424733998/_build_env/include/openssl/err.h:485:30: note: declared here
  485 | OSSL_DEPRECATEDIN_1_1_0 void ERR_remove_thread_state(void *);
      |                              ^~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [CMakeFiles/aziotsharedutil.dir/build.make:790: CMakeFiles/aziotsharedutil.dir/adapters/tlsio_openssl.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:907: CMakeFiles/aziotsharedutil.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

**Comment by jjerphan** [comment]
> However, I don't want you to waste time debugging the same errors that I already fixed ([#4 (comment)](https://github.com/conda-forge/azure-c-shared-utility-feedstock/pull/4#issuecomment-1603065392)). As @xylar requested above, I started working on this on my fork

Thanks for mentioning that. I think opening a draft PR would allow discoverability, even if it's still very WIP.

I do have time to look into the errors now, but I might in a couple of weeks.

**Comment by jdblischak** [comment]
> Thanks for mentioning that. I think opening a draft PR would allow discoverability, even if it's still very WIP.

Good idea. I just opened PR #5 to add the osx build. Once that is merged, I'll open another one with the linux build for us to troubleshoot

**Comment by xylar** [comment]
I will close this in favor of #5 and a follow-up Linux PR.

**Pull request closed without merging**